### PR TITLE
Ensure that netcore50 won't be broken by directly referencing S.P.CoreLib in facades

### PR DESCRIPTION
**Pull request by jkotas**
- Update System.Diagnostics.StackTrace to netstandard1.7
- Disable netcore50 builds that would get broken

### CODE REVIEW

**Comment by jkotas** [comment]
Prep-work for https://github.com/dotnet/corefx/pull/12187

**Comment by jkotas** [comment]
@weshaggard PTLA

File: src/System.Globalization/src/project.json [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-      "dependencies": {
-        "Microsoft.TargetingPack.Private.CoreCLR": "1.2.0-beta-24530-04"
-      },
-      "imports": [
-        "dotnet5.4"
-      ]
-    },
     "netstandard1.7": {
       "dependencies": {
         "Microsoft.TargetingPack.Private.CoreCLR": "1.2.0-beta-24530-04"
```
Comment (weshaggard) [inline src/System.Globalization/src/project.json]: Can remove the netcore50 section as well.

File: src/System.Reflection.Emit.ILGeneration/src/System.Reflection.Emit.ILGeneration.builds [hunk 2]
```diff
@@ -7,9 +7,6 @@
     <Project Include="System.Reflection.Emit.ILGeneration.csproj">
```
Comment (weshaggard) [inline src/System.Reflection.Emit.ILGeneration/src/System.Reflection.Emit.ILGeneration.builds]: Feel free to delete the commented out net46 configuration.

File: src/System.Reflection.Emit.Lightweight/src/System.Reflection.Emit.Lightweight.builds [hunk 3]
```diff
@@ -7,9 +7,6 @@
     <Project Include="System.Reflection.Emit.Lightweight.csproj">
```
Comment (weshaggard) [inline src/System.Reflection.Emit.Lightweight/src/System.Reflection.Emit.Lightweight.builds]: ditto for this net46 config.

File: src/System.Reflection.Emit/src/System.Reflection.Emit.builds [hunk 4]
```diff
@@ -7,9 +7,6 @@
     <Project Include="System.Reflection.Emit.csproj">
       <TargetGroup>net46</TargetGroup>
```
Comment (weshaggard) [inline src/System.Reflection.Emit/src/System.Reflection.Emit.builds]: ditto.

File: src/System.Resources.ResourceManager/src/System.Resources.ResourceManager.builds [hunk 5]
```diff
@@ -9,9 +9,6 @@
     </Project> -->
```
Comment (weshaggard) [inline src/System.Resources.ResourceManager/src/System.Resources.ResourceManager.builds]: I would clean out the other commented out builds here (net46 and netcore50aot).

**Comment by jkotas** [comment]
SegFault in System.IO.FileSystem.Watcher.Tests on OSX. Opened #12276

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test  please

**Pull request merged**